multimodal/src/autogluon/multimodal/utils/environment.py

```diff
@@ -322,7 +322,11 @@ def get_available_devices(num_gpus: int, auto_select_gpus: bool):
     -------
     The available devices.
     """
-    if num_gpus > 0:
+    if num_gpus == 1:
+        # for single GPU just return GPU 0,
+        # find_usable_cuda_devices can cause error if there is previous DDP run
```

Comment:
What's the reason of error?

Reply:
I think it's still because there's extra GPU usage in unused GPUs in previous DDP run. And future access to other GPUs after the DDP run will be blocked.